libraries/default.rb

```diff
@@ -173,6 +173,19 @@ def get_storage_info
       )
     end
 
+    def get_storage_hostname
+      storage_info = get_storage_info
+
+      if storage_info[:type] == 'external'
```

Comment:
Assuming it's not a problem that the `:type` symbol is not defined in the default return from get_storage_info.